### ISSUE TITLE
reduce asset name length limit

### ISSUE THREAD

**Issue by zoedberg** [LNP-BP/LNPBPs#74]
Currently there's an [enforcement at schema level](https://github.com/LNP-BP/rgb-node/blob/68146c578190bb578f2a4bf0f8d3298f94f0959a/src/contracts/fungible/data/schema.rs#L303) of `256` chars on the asset name.

This limit may be too high, so let's consider lowering it a bit.

**Comment by fedsten** [comment]
Which potential issues do you see by keeping the limit at the current level?

**Comment by zoedberg** [comment]
> Which potential issues do you see by keeping the limit at the current level?

I think that usually asset names will be short (e.g. ticker: "USDT" -> name: "USD Tether").

IMO allowing `256` chars can't cause dangerous issues, but from an interface perspective, every developer in every software that wants to show that info, will have to consider that potentially someone created an asset with a very long name. That's not a funny thing to deal with, especially if most of asset names are short and just one isn't.

So I think we need a reason to keep it so long, not the opposite.

My suggestion is to use `32` chars, it should be enough.

**Comment by fedsten** [comment]
Fair point, I agree with the proposal of reducing the chars limit to `32 `

**Comment by St333p** [comment]
I agree, a 32 chars limit helps to clarify the intended purpose of the "name" field.

**Comment by weedcoder** [comment]
in liquid, there is no asset name.
if you want to have a name, you have to register against blockstream registry and have some cryptographic proofs in a https served .well-know files.
you can assign asset name manually in the client configuration file, not really practical.
here is the documentation : https://docs.blockstream.com/liquid/developer-guide/proof-of-issuance.html

**Comment by dieeasy** [comment]
- the longest currency name in wikipedia's [List of circulating currencies](https://en.wikipedia.org/wiki/List_of_circulating_currencies) is `Bosnia and Herzegovina convertible mark` with **39** characters
- the longest ERC20 token name (according to [etherscan.io](https://etherscan.io/token/) is [Yee](https://etherscan.io/token/0x922105fad8153f516bcfb829f56dc097a0e1d705) with **46** characters (`Yee - A Blockchain-powered & Cloud-based Socia`) or **40** when ignoring the leading `Yee - ` (also, looks like it's been truncated)

the answer might be 42 :smirk:

**Comment by dr-orlovsky** [comment]
I think we need 37

![image](https://user-images.githubusercontent.com/372034/99862412-ef513780-2b99-11eb-9cf3-3807a35444c8.jpeg)
